packages/core/upload/server/src/services/extensions/index.ts

```diff
@@ -1,7 +1,53 @@
-import * as contentManager from './content-manager';
-import * as core from './core';
+import { async } from '@strapi/utils';
+import { signEntityMedia } from './utils';
+
+const signFileUrlsOnDocumentService = async () => {
+  const { provider } = strapi.plugins.upload;
+  const isPrivate = await provider.isPrivate();
+
+  // We only need to sign the file urls if the provider is private
+  if (!isPrivate) {
+    return;
+  }
+
+  strapi.documents.use(async (ctx, next) => {
+    const uid = ctx.uid;
+    const result: any = await next();
+
+    if (ctx.action === 'findMany') {
```

Comment:
Does a switch statement make more sense here? totally up to you

Reply:
I personally don't mind, was using if cases because it's what we have in the other document service middlewares